src/server/sams/api/produce/assets.py

```diff
@@ -22,19 +22,105 @@
 **schema**               :attr:`sams_client.schemas.assets.ASSET_SCHEMA`
 =====================   =================================================================
 """
+from bson import ObjectId
 from flask import current_app as app, json
+from flask import request
 
 from sams.api.service import SamsApiService
 from sams.api.consume import ConsumeAssetResource
 from sams_client.schemas import SET_STATES
 from sams.sets import get_service as get_sets_service
+from sams.assets import get_service as get_asset_service
 from sams.storage.sams_media_storage import get_request_id
 from superdesk.resource import Resource, build_custom_hateoas
 from sams_client.errors import SamsAssetErrors
+from sams.utils import get_external_user_id, get_external_session_id
 
+from superdesk import Blueprint
+from superdesk.utc import utcnow
 
 FIELDS_TO_JSON_PARSE = ('tags', 'extra')
 
+assets_produce_bp = Blueprint('assets_produce', __name__)
+
+
+@assets_produce_bp.route('/produce/assets/lock/<asset_id>', methods=['PATCH'])
+def lock_asset(asset_id: str):
+    """
+    Uses asset_id and lock action and locks the corresponding asset
+    """
+
+    service = get_asset_service()
+    asset = service.get_by_id(asset_id)
+    updates = {}
+
+    try:
+        lock_action = asset['lock_action']
+    except KeyError:
+        lock_action = None
+        pass
+
+    if lock_action:
+        raise SamsAssetErrors.LockingAssetLocked
```

Comment:
All SAMS exceptions need to be called when raising them, i.e: `raise SamsAssetErrors.LockingAssetLocked()`